util/a2e/atoe_utils.c

```diff
@@ -545,22 +548,29 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
+					char *tempPattern;
 
-					b = a2e((char *)pattern, fmt - pattern);
+					tempPattern = a2e((char *)pattern, fmt - pattern);
 
 					/* Extract a double from args, this works for both doubles
 					 * and floats,
 					 * NB if we use float for a single precision floating
 					 * point number the result is wrong.
 					 */
-					len = sprintf(this.buffer, b, va_arg(args, double));
-					free(b);
-					b = e2a_string(this.buffer);
-					strcpy(this.buffer, b);
-					free(b);
-					this.buffer += len;
+					size_t freeCap = count > this.counter ? count - this.counter : 0;
```

Comment:
For C code, variables need to declared at the beginning of scope. https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#variable-declarations

Reply:
Just wanted to double check.  vsnprintf writes `count` minus 1 characters to account for null-terminating character.  Should we saturate at `count - 1` instead?

Reply:
We do not count the termination char when putting it at the end of the buffer therefore the return value would be correct.